<h3>1단계</h3>
<pre>
두 자연수 A와 B가 주어진다. 이때, A+B, A-B, A*B, A/B(몫), A%B(나머지)를 출력하는 프로그램을 작성하시오. 
</pre>

In [ ]:
A, B = map(int, input().split( ))
print(A+B)
print(A-B)
print(A*B)
print(A//B)
print(A%B)

<pre>
입력
첫째 줄에 준하가 가입하려고 하는 사이트에 이미 존재하는 아이디가 주어진다. 아이디는 알파벳 소문자로만 이루어져 있으며, 길이는 50자를 넘지 않는다.

출력
첫째 줄에 준하의 놀람을 출력한다. 놀람은 아이디 뒤에 ??!를 붙여서 나타낸다.
</pre>

In [ ]:
print(input(), "??!", sep="")

<pre>
문제
ICPC Bangkok Regional에 참가하기 위해 수완나품 국제공항에 막 도착한 팀 레드시프트 일행은 눈을 믿을 수 없었다. 공항의 대형 스크린에 올해가 2562년이라고 적혀 있던 것이었다.

불교 국가인 태국은 불멸기원(佛滅紀元), 즉 석가모니가 열반한 해를 기준으로 연도를 세는 불기를 사용한다. 반면, 우리나라는 서기 연도를 사용하고 있다. 불기 연도가 주어질 때 이를 서기 연도로 바꿔 주는 프로그램을 작성하시오.

입력
서기 연도를 알아보고 싶은 불기 연도 y가 주어진다. (1000 ≤ y ≤ 3000)

출력
불기 연도를 서기 연도로 변환한 결과를 출력한다.
</pre>

In [ ]:
print(int(input())-543)

<pre>
입력
첫째 줄에 A, B, C가 순서대로 주어진다. (2 ≤ A, B, C ≤ 10000)

출력
첫째 줄에 (A+B)%C, 둘째 줄에 ((A%C) + (B%C))%C, 셋째 줄에 (A×B)%C, 넷째 줄에 ((A%C) × (B%C))%C를 출력한다.
</pre>


In [ ]:
A, B, C = map(int, input().split( ))
print((A+B)%C)
print(((A%C) + (B%C))%C)
print((A*B)%C)
print(((A%C) * (B%C))%C)


<pre>
(세 자리 수) × (세 자리 수) 곱셈의 중간 과정과 결과
</pre>

In [ ]:
# 472 385
A = int(input())
B = int(input())
a = B%10
b = B%100//10
c = B//100

print(A*a)
print(A*b)
print(A*c)
print(A*B)

<pre>
입력
첫 번째 줄에 A, B, C (1 ≤ A, B, C ≤ 1012)이 공백을 사이에 두고 주어진다.

출력
A+B+C의 값을 출력한다.
</pre>

In [ ]:
A, B, C = map(int, input().split( ))
print(A+B+C)

<pre>
고양이 출력하기

\    /\
 )  ( ')
(  /  )
 \(__)|
</pre>

In [ ]:
cat = '''\\    /\\
 )  ( ')
(  /  )
 \\(__)|'''
print(cat)

<pre>
강아지 출력
|\_/|
|q p|   /}
( 0 )"""\
|"^"`    |
||_/=\\__|
</pre>

In [ ]:
dog = """|\\_/|
|q p|   /}
( 0 )\"\"\"\\
|"^"`    |
||_/=\\\\__|"""
print(dog)

<h3>2단계</h3>
<pre>
문제
두 정수 A와 B가 주어졌을 때, A와 B를 비교하는 프로그램을 작성하시오.

입력
첫째 줄에 A와 B가 주어진다. A와 B는 공백 한 칸으로 구분되어져 있다.

출력
첫째 줄에 다음 세 가지 중 하나를 출력한다.

A가 B보다 큰 경우에는 '>'를 출력한다.
A가 B보다 작은 경우에는 '<'를 출력한다.
A와 B가 같은 경우에는 '=='를 출력한다.
</pre>

In [ ]:
A, B = map(int, input().split( ))

if A>B:
    print(">")
elif A<B:
    print("<")
else:
    print("==")

<pre>
문제
시험 점수를 입력받아 90 ~ 100점은 A, 80 ~ 89점은 B, 70 ~ 79점은 C, 60 ~ 69점은 D, 나머지 점수는 F를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 시험 점수가 주어진다. 시험 점수는 0보다 크거나 같고, 100보다 작거나 같은 정수이다.

출력
시험 성적을 출력한다.
</pre>

In [ ]:
x = int(input())

if 90 <= x <= 100:
    print("A")
elif 80 <= x <= 89:
    print("B")
elif 70 <= x <= 79:
    print("C")
elif 60 <= x <= 69:
    print("D")
else:
    print("F")

<pre>
문제
연도가 주어졌을 때, 윤년이면 1, 아니면 0을 출력하는 프로그램을 작성하시오.

윤년은 연도가 4의 배수이면서, 100의 배수가 아닐 때 또는 400의 배수일 때이다.

예를 들어, 2012년은 4의 배수이면서 100의 배수가 아니라서 윤년이다. 1900년은 100의 배수이고 400의 배수는 아니기 때문에 윤년이 아니다. 하지만, 2000년은 400의 배수이기 때문에 윤년이다.

입력
첫째 줄에 연도가 주어진다. 연도는 1보다 크거나 같고, 4000보다 작거나 같은 자연수이다.

출력
첫째 줄에 윤년이면 1, 아니면 0을 출력한다.
</pre>

In [ ]:
year = int(input())

if (year%4==0 and not year%100==0) or year%400==0:
    print(1)
else:
    print(0)

<pre>
입력
첫 줄에는 정수 x가 주어진다. (−1000 ≤ x ≤ 1000; x ≠ 0) 다음 줄에는 정수 y가 주어진다. (−1000 ≤ y ≤ 1000; y ≠ 0)

출력
점 (x, y)의 사분면 번호(1, 2, 3, 4 중 하나)를 출력한다.
</pre>

In [ ]:
x = int(input())
y = int(input())

if x>0 and y>0:
    print(1)
elif x<0 and y>0:
    print(2)
elif x<0 and y<0:
    print(3)
else:
    print(4)

<pre>
입력
첫째 줄에 두 정수 H와 M이 주어진다. (0 ≤ H ≤ 23, 0 ≤ M ≤ 59) 그리고 이것은 현재 상근이가 설정한 알람 시간 H시 M분을 의미한다.

입력 시간은 24시간 표현을 사용한다. 24시간 표현에서 하루의 시작은 0:0(자정)이고, 끝은 23:59(다음날 자정 1분 전)이다. 시간을 나타낼 때, 불필요한 0은 사용하지 않는다.

출력
첫째 줄에 상근이가 창영이의 방법을 사용할 때, 설정해야 하는 알람 시간을 출력한다. (입력과 같은 형태로 출력하면 된다.)
</pre>

In [ ]:
H, M = map(int, input().split( ))
T = 45
if M < T:
    M = 60 + ( M - T)
    if H == 0:
        H = 23
    else:
        H -= 1
else:
    M -= 45

print(H, M)

<pre>
입력
첫째 줄에는 현재 시각이 나온다. 현재 시각은 시 A (0 ≤ A ≤ 23) 와 분 B (0 ≤ B ≤ 59)가 정수로 빈칸을 사이에 두고 순서대로 주어진다. 두 번째 줄에는 요리하는 데 필요한 시간 C (0 ≤ C ≤ 1,000)가 분 단위로 주어진다.

출력
첫째 줄에 종료되는 시각의 시와 분을 공백을 사이에 두고 출력한다. (단, 시는 0부터 23까지의 정수, 분은 0부터 59까지의 정수이다. 디지털 시계는 23시 59분에서 1분이 지나면 0시 0분이 된다.)
</pre>

In [ ]:
A, B = map(int, input().split( ))
C = int(input())
H = (B+C)//60
print((A+H)%24, (B+C)%60)

<pre>
문제
1에서부터 6까지의 눈을 가진 3개의 주사위를 던져서 다음과 같은 규칙에 따라 상금을 받는 게임이 있다.

같은 눈이 3개가 나오면 10,000원+(같은 눈)×1,000원의 상금을 받게 된다.
같은 눈이 2개만 나오는 경우에는 1,000원+(같은 눈)×100원의 상금을 받게 된다.
모두 다른 눈이 나오는 경우에는 (그 중 가장 큰 눈)×100원의 상금을 받게 된다.
예를 들어, 3개의 눈 3, 3, 6이 주어지면 상금은 1,000+3×100으로 계산되어 1,300원을 받게 된다. 
또 3개의 눈이 2, 2, 2로 주어지면 10,000+2×1,000 으로 계산되어 12,000원을 받게 된다. 
3개의 눈이 6, 2, 5로 주어지면 그중 가장 큰 값이 6이므로 6×100으로 계산되어 600원을 상금으로 받게 된다.

3개 주사위의 나온 눈이 주어질 때, 상금을 계산하는 프로그램을 작성 하시오.

입력
첫째 줄에 3개의 눈이 빈칸을 사이에 두고 각각 주어진다.

출력
첫째 줄에 게임의 상금을 출력 한다.
</pre>

In [ ]:
A, B, C = map(int, input().split( ))
isSame = [A==B, A==C, B==C]

# 모든 눈이 같을 때
if sum(isSame)==3:
    print(10000+A*1000)
# 두 눈이 같을 때
elif sum(isSame)==1:
    # 첫번쨰 눈과 두번째나 세번째가 같을 때
    if isSame[0] == 1 or isSame[1] == 1:
        print(1000+A*100)
    # 두번째와 세번쨰 눈이 같을 때
    else:
        print(1000+B*100)
# 다 다를 때
else:
    X = max(A, B, C)
    print(X*100)

<h3>3단계</h3>
<pre>
문제
N을 입력받은 뒤, 구구단 N단을 출력하는 프로그램을 작성하시오. 출력 형식에 맞춰서 출력하면 된다.

입력
첫째 줄에 N이 주어진다. N은 1보다 크거나 같고, 9보다 작거나 같다.

출력
출력형식과 같게 N*1부터 N*9까지 출력한다.
</pre>

In [ ]:
N = int(input())
for i in range(9):
    print(N, '*', i+1, '=', N*(i+1))

<pre>
문제
두 정수 A와 B를 입력받은 다음, A+B를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 테스트 케이스의 개수 T가 주어진다.

각 테스트 케이스는 한 줄로 이루어져 있으며, 각 줄에 A와 B가 주어진다. (0 < A, B < 10)

출력
각 테스트 케이스마다 A+B를 출력한다.
</pre>

In [ ]:
T = int(input())
for i in range(T):
    A, B = map(int, input().split( ))
    print(A+B)


<pre>
문제
n이 주어졌을 때, 1부터 n까지 합을 구하는 프로그램을 작성하시오.

입력
첫째 줄에 n (1 ≤ n ≤ 10,000)이 주어진다.

출력
1부터 n까지 합을 출력한다.
</pre>

In [ ]:
n = int(input())
print(n*(1+n)//2)

<pre>
문제
준원이는 저번 주에 살면서 처음으로 코스트코를 가 봤다. 정말 멋졌다. 그런데, 몇 개 담지도 않았는데 수상하게 높은 금액이 나오는 것이다! 준원이는 영수증을 보면서 정확하게 계산된 것이 맞는지 확인해보려 한다.

영수증에 적힌,

구매한 각 물건의 가격과 개수
구매한 물건들의 총 금액
을 보고, 구매한 물건의 가격과 개수로 계산한 총 금액이 영수증에 적힌 총 금액과 일치하는지 검사해보자.

입력
첫째 줄에는 영수증에 적힌 총 금액 
$X$가 주어진다.

둘째 줄에는 영수증에 적힌 구매한 물건의 종류의 수 
$N$이 주어진다.

이후 
$N$개의 줄에는 각 물건의 가격 
$a$와 개수 
$b$가 공백을 사이에 두고 주어진다.

출력
구매한 물건의 가격과 개수로 계산한 총 금액이 영수증에 적힌 총 금액과 일치하면 Yes를 출력한다. 일치하지 않는다면 No를 출력한다.
</pre>

In [ ]:
X = int(input())
N = int(input())

price = 0
for i in range(N):
    a, b = map(int, input().split( ))
    price += a*b

if price == X:
    print("Yes")
else:
    print("No")

입력
첫 번째 줄에는 문제의 정수 
$N$이 주어진다. 
$(4\le N\le 1\, 000$; 
$N$은 
$4$의 배수
$)$ 

출력
혜아가 
$N$바이트 정수까지 저장할 수 있다고 생각하는 정수 자료형의 이름을 출력하여라.

In [ ]:
N = int(input())
print(" ".join("long" for i in range(N//4)), 'int')

<pre>
본격적으로 for문 문제를 풀기 전에 주의해야 할 점이 있다. 입출력 방식이 느리면 여러 줄을 입력받거나 출력할 때 시간초과가 날 수 있다는 점이다.

C++을 사용하고 있고 cin/cout을 사용하고자 한다면, cin.tie(NULL)과 sync_with_stdio(false)를 둘 다 적용해 주고, endl 대신 개행문자(\n)를 쓰자. 단, 이렇게 하면 더 이상 scanf/printf/puts/getchar/putchar 등 C의 입출력 방식을 사용하면 안 된다.

Java를 사용하고 있다면, Scanner와 System.out.println 대신 BufferedReader와 BufferedWriter를 사용할 수 있다. BufferedWriter.flush는 맨 마지막에 한 번만 하면 된다.

Python을 사용하고 있다면, input 대신 sys.stdin.readline을 사용할 수 있다. 단, 이때는 맨 끝의 개행문자까지 같이 입력받기 때문에 문자열을 저장하고 싶을 경우 .rstrip()을 추가로 해 주는 것이 좋다.

또한 입력과 출력 스트림은 별개이므로, 테스트케이스를 전부 입력받아서 저장한 뒤 전부 출력할 필요는 없다. 테스트케이스를 하나 받은 뒤 하나 출력해도 된다.

자세한 설명 및 다른 언어의 경우는 이 글에 설명되어 있다.

이 블로그 글에서 BOJ의 기타 여러 가지 팁을 볼 수 있다.

입력
첫 줄에 테스트케이스의 개수 T가 주어진다. T는 최대 1,000,000이다. 다음 T줄에는 각각 두 정수 A와 B가 주어진다. A와 B는 1 이상, 1,000 이하이다.

출력
각 테스트케이스마다 A+B를 한 줄에 하나씩 순서대로 출력한다.
</pre>

In [ ]:
import sys

T = int(sys.stdin.readline().rstrip())
for i in range(T):
    a, b = map(int, sys.stdin.readline().rstrip().split())
    print(a+b)

In [ ]:
T = int(input())
for i in range(T):
    a, b = map(int, input().split())
    print(a+b)

<pre>
문제
두 정수 A와 B를 입력받은 다음, A+B를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 테스트 케이스의 개수 T가 주어진다.

각 테스트 케이스는 한 줄로 이루어져 있으며, 각 줄에 A와 B가 주어진다. (0 < A, B < 10)

출력
각 테스트 케이스마다 "Case #x: "를 출력한 다음, A+B를 출력한다. 테스트 케이스 번호는 1부터 시작한다.
</pre>

In [ ]:
T = int(input())
for i in range(T):
    A, B = map(int, input().split())
    print(f"Case #{i+1}: {A+B}")

<pre>
문제
두 정수 A와 B를 입력받은 다음, A+B를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 테스트 케이스의 개수 T가 주어진다.

각 테스트 케이스는 한 줄로 이루어져 있으며, 각 줄에 A와 B가 주어진다. (0 < A, B < 10)

출력
각 테스트 케이스마다 "Case #x: A + B = C" 형식으로 출력한다. x는 테스트 케이스 번호이고 1부터 시작하며, C는 A+B이다.
</pre>

In [ ]:
T = int(input())
for i in range(T):
    A, B = map(int, input().split())
    print(f"Case #{i+1}: {A} + {B} = {A+B}")

<pre>
문제
첫째 줄에는 별 1개, 둘째 줄에는 별 2개, N번째 줄에는 별 N개를 찍는 문제

입력
첫째 줄에 N(1 ≤ N ≤ 100)이 주어진다.

출력
첫째 줄부터 N번째 줄까지 차례대로 별을 출력한다.
</pre>

In [ ]:
N = int(input())
for i in range(1, N+1):
    print("".join(" " for x in range(0, N-i)), end='')
    print("".join("*" for x in range(N-i, N)))

<pre>
문제
두 정수 A와 B를 입력받은 다음, A+B를 출력하는 프로그램을 작성하시오.

입력
입력은 여러 개의 테스트 케이스로 이루어져 있다.

각 테스트 케이스는 한 줄로 이루어져 있으며, 각 줄에 A와 B가 주어진다. (0 < A, B < 10)

입력의 마지막에는 0 두 개가 들어온다.

출력
각 테스트 케이스마다 A+B를 출력한다.
</pre>

In [ ]:
A, B = True, True
while True:
    A, B = map(int, input().split())
    if A and B:
        print(A+B)
    else:
        break

<pre>
문제
두 정수 A와 B를 입력받은 다음, A+B를 출력하는 프로그램을 작성하시오.

입력
입력은 여러 개의 테스트 케이스로 이루어져 있다.

각 테스트 케이스는 한 줄로 이루어져 있으며, 각 줄에 A와 B가 주어진다. (0 < A, B < 10)

출력
각 테스트 케이스마다 A+B를 출력한다.
</pre>

In [ ]:
while True:
    try:
        A, B = map(int, input().split())
        print(A+B)
    except:
        break

<h3>4단계</h3>
<pre>
문제
총 N개의 정수가 주어졌을 때, 정수 v가 몇 개인지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 정수의 개수 N(1 ≤ N ≤ 100)이 주어진다. 둘째 줄에는 정수가 공백으로 구분되어져있다. 셋째 줄에는 찾으려고 하는 정수 v가 주어진다. 입력으로 주어지는 정수와 v는 -100보다 크거나 같으며, 100보다 작거나 같다.

출력
첫째 줄에 입력으로 주어진 N개의 정수 중에 v가 몇 개인지 출력한다.
</pre>

In [ ]:
N = int(input())
L = list(map(int, input().split()))
v = int(input())

n = 0
for i in L:
    if i == v:
        n += 1
        
print(n)

<pre>
문제
정수 N개로 이루어진 수열 A와 정수 X가 주어진다. 이때, A에서 X보다 작은 수를 모두 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 N과 X가 주어진다. (1 ≤ N, X ≤ 10,000)

둘째 줄에 수열 A를 이루는 정수 N개가 주어진다. 주어지는 정수는 모두 1보다 크거나 같고, 10,000보다 작거나 같은 정수이다.

출력
X보다 작은 수를 입력받은 순서대로 공백으로 구분해 출력한다. X보다 작은 수는 적어도 하나 존재한다.

예제 입력 1 
10 5
1 10 4 9 2 3 8 5 7 6
예제 출력 1 
1 4 2 3
</pre>

In [ ]:
N, X = map(int, input().split())
L1 = list(map(int, input().split()))
L2 = []

for i in L1:
    if i < X:
        L2.append(i)
        
for i in L2:
    print(i, end=" ")

<pre>
문제
N개의 정수가 주어진다. 이때, 최솟값과 최댓값을 구하는 프로그램을 작성하시오.

입력
첫째 줄에 정수의 개수 N (1 ≤ N ≤ 1,000,000)이 주어진다. 둘째 줄에는 N개의 정수를 공백으로 구분해서 주어진다. 모든 정수는 -1,000,000보다 크거나 같고, 1,000,000보다 작거나 같은 정수이다.

출력
첫째 줄에 주어진 정수 N개의 최솟값과 최댓값을 공백으로 구분해 출력한다.

예제 입력 1 
5
20 10 35 30 7
예제 출력 1 
7 35
</pre>

In [ ]:
N = int(input())
L = list(map(int, input().split()))
print(min(L), max(L))

<pre>
문제
9개의 서로 다른 자연수가 주어질 때, 이들 중 최댓값을 찾고 그 최댓값이 몇 번째 수인지를 구하는 프로그램을 작성하시오.

예를 들어, 서로 다른 9개의 자연수

3, 29, 38, 12, 57, 74, 40, 85, 61

이 주어지면, 이들 중 최댓값은 85이고, 이 값은 8번째 수이다.

입력
첫째 줄부터 아홉 번째 줄까지 한 줄에 하나의 자연수가 주어진다. 주어지는 자연수는 100 보다 작다.

출력
첫째 줄에 최댓값을 출력하고, 둘째 줄에 최댓값이 몇 번째 수인지를 출력한다.

예제 입력 1 
3
29
38
12
57
74
40
85
61

예제 출력 1 
85
8
</pre>

In [ ]:
L = list(int(input()) for x in range(9))
m = max(L)
print(m)
print(L.index(m)+1)

<pre>
문제
도현이는 바구니를 총 N개 가지고 있고, 각각의 바구니에는 1번부터 N번까지 번호가 매겨져 있다. 또, 1번부터 N번까지 번호가 적혀있는 공을 매우 많이 가지고 있다. 가장 처음 바구니에는 공이 들어있지 않으며, 바구니에는 공을 1개만 넣을 수 있다.

도현이는 앞으로 M번 공을 넣으려고 한다. 도현이는 한 번 공을 넣을 때, 공을 넣을 바구니 범위를 정하고, 정한 바구니에 모두 같은 번호가 적혀있는 공을 넣는다. 만약, 바구니에 공이 이미 있는 경우에는 들어있는 공을 빼고, 새로 공을 넣는다. 공을 넣을 바구니는 연속되어 있어야 한다.

공을 어떻게 넣을지가 주어졌을 때, M번 공을 넣은 이후에 각 바구니에 어떤 공이 들어 있는지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 N (1 ≤ N ≤ 100)과 M (1 ≤ M ≤ 100)이 주어진다.

둘째 줄부터 M개의 줄에 걸쳐서 공을 넣는 방법이 주어진다. 각 방법은 세 정수 i j k로 이루어져 있으며, i번 바구니부터 j번 바구니까지에 k번 번호가 적혀져 있는 공을 넣는다는 뜻이다. 예를 들어, 2 5 6은 2번 바구니부터 5번 바구니까지에 6번 공을 넣는다는 뜻이다. (1 ≤ i ≤ j ≤ N, 1 ≤ k ≤ N)

도현이는 입력으로 주어진 순서대로 공을 넣는다.

출력
1번 바구니부터 N번 바구니에 들어있는 공의 번호를 공백으로 구분해 출력한다. 공이 들어있지 않은 바구니는 0을 출력한다.

예제 입력 1 
5 4
1 2 3
3 4 4
1 4 1
2 2 2
예제 출력 1 
1 2 1 1 0
</pre>

In [ ]:
N, M = map(int, input().split())
L = [0]*N
for x in range(M):
    i, j, k = map(int, input().split())
    for x in range(i-1, j):
        L[x] = k
for i in L:
    print(i, end=' ')

<pre>
문제
도현이는 바구니를 총 N개 가지고 있고, 각각의 바구니에는 1번부터 N번까지 번호가 매겨져 있다. 
바구니에는 공이 1개씩 들어있고, 처음에는 바구니에 적혀있는 번호와 같은 번호가 적힌 공이 들어있다.

도현이는 앞으로 M번 공을 바꾸려고 한다. 도현이는 공을 바꿀 바구니 2개를 선택하고, 두 바구니에 들어있는 공을 서로 교환한다.

공을 어떻게 바꿀지가 주어졌을 때, M번 공을 바꾼 이후에 각 바구니에 어떤 공이 들어있는지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 N (1 ≤ N ≤ 100)과 M (1 ≤ M ≤ 100)이 주어진다.

둘째 줄부터 M개의 줄에 걸쳐서 공을 교환할 방법이 주어진다. 각 방법은 두 정수 i j로 이루어져 있으며, i번 바구니와 j번 바구니에 들어있는 공을 교환한다는 뜻이다. (1 ≤ i ≤ j ≤ N)

도현이는 입력으로 주어진 순서대로 공을 교환한다.

출력
1번 바구니부터 N번 바구니에 들어있는 공의 번호를 공백으로 구분해 출력한다.

예제 입력 1 
5 4
1 2
3 4
1 4
2 2

예제 출력 1 
3 1 4 2 5
</pre>

In [ ]:
N, M = map(int, input().split())
L = [i+1 for i in range(N)]
for i in range(M):
    x, y = map(int, input().split())
    L[x-1], L[y-1] = L[y-1], L[x-1]
print(*L)

<pre>
문제
X대학 M교수님은 프로그래밍 수업을 맡고 있다. 교실엔 학생이 30명이 있는데, 학생 명부엔 각 학생별로 1번부터 30번까지 출석번호가 붙어 있다.

교수님이 내준 특별과제를 28명이 제출했는데, 그 중에서 제출 안 한 학생 2명의 출석번호를 구하는 프로그램을 작성하시오.

입력
입력은 총 28줄로 각 제출자(학생)의 출석번호 n(1 ≤ n ≤ 30)가 한 줄에 하나씩 주어진다. 출석번호에 중복은 없다.

출력
출력은 2줄이다. 1번째 줄엔 제출하지 않은 학생의 출석번호 중 가장 작은 것을 출력하고, 2번째 줄에선 그 다음 출석번호를 출력한다.
</pre>

In [ ]:
L = [i for i in range(1, 31)]

for i in range(28):
    L.remove(int(input()))

print(min(L))
print(max(L))

<pre>
문제
두 자연수 A와 B가 있을 때, A%B는 A를 B로 나눈 나머지 이다. 예를 들어, 7, 14, 27, 38을 3으로 나눈 나머지는 1, 2, 0, 2이다. 

수 10개를 입력받은 뒤, 이를 42로 나눈 나머지를 구한다. 그 다음 서로 다른 값이 몇 개 있는지 출력하는 프로그램을 작성하시오.

입력
첫째 줄부터 열번째 줄 까지 숫자가 한 줄에 하나씩 주어진다. 이 숫자는 1,000보다 작거나 같고, 음이 아닌 정수이다.

출력
첫째 줄에, 42로 나누었을 때, 서로 다른 나머지가 몇 개 있는지 출력한다.
</pre>

In [ ]:

print(len({i%42 for i in [int(input()) for i in range(10)]}))

<pre>
문제
도현이는 바구니를 총 N개 가지고 있고, 각각의 바구니에는 1번부터 N번까지 번호가 순서대로 적혀져 있다.
바구니는 일렬로 놓여져 있고, 가장 왼쪽 바구니를 1번째 바구니, 그 다음 바구니를 2번째 바구니, ..., 가장 오른쪽 바구니를 N번째 바구니라고 부른다. 

도현이는 앞으로 M번 바구니의 순서를 역순으로 만들려고 한다. 도현이는 한 번 순서를 역순으로 바꿀 때, 순서를 역순으로 만들 범위를 정하고, 그 범위에 들어있는 바구니의 순서를 역순으로 만든다.

바구니의 순서를 어떻게 바꿀지 주어졌을 때, M번 바구니의 순서를 역순으로 만든 다음, 바구니에 적혀있는 번호를 가장 왼쪽 바구니부터 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 N (1 ≤ N ≤ 100)과 M (1 ≤ M ≤ 100)이 주어진다.

둘째 줄부터 M개의 줄에는 바구니의 순서를 역순으로 만드는 방법이 주어진다. 방법은 i j로 나타내고, 왼쪽으로부터 i번째 바구니부터 j번째 바구니의 순서를 역순으로 만든다는 뜻이다. (1 ≤ i ≤ j ≤ N)

도현이는 입력으로 주어진 순서대로 바구니의 순서를 바꾼다.

출력
모든 순서를 바꾼 다음에, 가장 왼쪽에 있는 바구니부터 바구니에 적혀있는 순서를 공백으로 구분해 출력한다.

5 4
1 2
3 4
1 4
2 2


예제 출력 1 
3 4 1 2 5
</pre>

In [ ]:
N, M = map(int, input().split())
L = [i for i in range(1, N+1)]

for x in range(M):
    i, j = map(lambda x: int(x) - 1, input().split())
    L = L[:i]+list(reversed(L[i:j+1]))+L[j+1:]
print(*L)

In [ ]:
L = [i for i in range(1, 11)]
L2 = list(reversed(L[2:5]))
print(L[:2])
print(L2)
print(L[5:])
L = L[:2]+L2+L[5:]
print(L)

L2 = list(reversed(L[2:5]))
print(L[:2])
print(L2)
print(L[5:])
L = L[:2]+L2+L[5:]
print(L)

In [ ]:
N, M = 5, 4
L = [i for i in range(1, N+1)]


i, j = 1, 2
i -= 1
j -= 1
L = L[:i]+list(reversed(L[i:j+1]))+L[j+1:]
print(list(reversed(L[i:j+1])))
print(L)

<pre>
문제
세준이는 기말고사를 망쳤다. 세준이는 점수를 조작해서 집에 가져가기로 했다. 일단 세준이는 자기 점수 중에 최댓값을 골랐다. 이 값을 M이라고 한다. 그리고 나서 모든 점수를 점수/M*100으로 고쳤다.

예를 들어, 세준이의 최고점이 70이고, 수학점수가 50이었으면 수학점수는 50/70*100이 되어 71.43점이 된다.

세준이의 성적을 위의 방법대로 새로 계산했을 때, 새로운 평균을 구하는 프로그램을 작성하시오.

입력
첫째 줄에 시험 본 과목의 개수 N이 주어진다. 이 값은 1000보다 작거나 같다. 둘째 줄에 세준이의 현재 성적이 주어진다. 이 값은 100보다 작거나 같은 음이 아닌 정수이고, 적어도 하나의 값은 0보다 크다.

출력
첫째 줄에 새로운 평균을 출력한다. 실제 정답과 출력값의 절대오차 또는 상대오차가 10-2 이하이면 정답이다.

예제 입력 1 
3
40 80 60

예제 출력 1 
75.0
</pre>

In [ ]:
N = int(input())
L = list(map(int, input().split()))
M = max(L)
for i in range(len(L)):
    L[i] = L[i]/M*100
print(sum(L)/len(L))

<pre>
<h3>문자열</h3>
문제
단어 
S와 정수 
i가 주어졌을 때, 
S의 
i번째 글자를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 영어 소문자와 대문자로만 이루어진 단어 
S가 주어진다. 단어의 길이는 최대 
1\,000이다.

둘째 줄에 정수 i가 주어진다. (
$1 \le i \le \left|S\right|$)

출력
 
S의 
i번째 글자를 출력한다.

예제 입력 1 
Sprout
3
예제 출력 1 
r
</pre>

In [ ]:
print(input()[int(input())-1])

<pre>
문제
알파벳으로만 이루어진 단어를 입력받아, 그 길이를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 영어 소문자와 대문자로만 이루어진 단어가 주어진다. 단어의 길이는 최대 100이다.

출력
첫째 줄에 입력으로 주어진 단어의 길이를 출력한다.
</pre>

In [ ]:
print(len(input()))

<pre>
문제
문자열을 입력으로 주면 문자열의 첫 글자와 마지막 글자를 출력하는 프로그램을 작성하시오.

입력
입력의 첫 줄에는 테스트 케이스의 개수 T(1 ≤ T ≤ 10)가 주어진다. 각 테스트 케이스는 한 줄에 하나의 문자열이 주어진다. 문자열은 알파벳 A~Z 대문자로 이루어지며 알파벳 사이에 공백은 없으며 문자열의 길이는 1000보다 작다.

출력
각 테스트 케이스에 대해서 주어진 문자열의 첫 글자와 마지막 글자를 연속하여 출력한다.

예제 입력 1 
3
ACDKJFOWIEGHE
O
AB

예제 출력 1 
AE
OO
AB
</pre>

In [ ]:
T = int(input())

for i in range(T):
    S = input()
    print(S[0]+S[len(S)-1])
    

<pre>
문제
알파벳 소문자, 대문자, 숫자 0-9중 하나가 주어졌을 때, 주어진 글자의 아스키 코드값을 출력하는 프로그램을 작성하시오.

입력
알파벳 소문자, 대문자, 숫자 0-9 중 하나가 첫째 줄에 주어진다.

출력
입력으로 주어진 글자의 아스키 코드 값을 출력한다.
</pre>

In [ ]:
# ord란 ordinal position의 약자로, 문자의 순서 위치 값을 의미 안녕 dgktpd d drrmfj그러나 
print(ord(input()))

<pre>
N개의 숫자가 공백 없이 쓰여있다. 이 숫자를 모두 합해서 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 숫자의 개수 N (1 ≤ N ≤ 100)이 주어진다. 둘째 줄에 숫자 N개가 공백없이 주어진다.

출력
입력으로 주어진 숫자 N개의 합을 출력한다.

예제 입력 1 
1
1
예제 출력 1 
1
예제 입력 2 
5
54321
예제 출력 2 
</pre>

In [ ]:
N = int(input())
S = input()
n = 0
for i in range(N):
    n += int(S[i])
print(n)

<pre>
문제
알파벳 소문자로만 이루어진 단어 S가 주어진다. 
각각의 알파벳에 대해서, 단어에 포함되어 있는 경우에는 처음 등장하는 위치를, 포함되어 있지 않은 경우에는 -1을 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 단어 S가 주어진다. 단어의 길이는 100을 넘지 않으며, 알파벳 소문자로만 이루어져 있다.

출력
각각의 알파벳에 대해서, a가 처음 등장하는 위치, b가 처음 등장하는 위치, ... z가 처음 등장하는 위치를 공백으로 구분해서 출력한다.

만약, 어떤 알파벳이 단어에 포함되어 있지 않다면 -1을 출력한다. 단어의 첫 번째 글자는 0번째 위치이고, 두 번째 글자는 1번째 위치이다.

예제 입력 1 
baekjoon

예제 출력 1 
1 0 -1 -1 2 -1 -1 -1 -1 4 3 -1 -1 7 5 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
</pre>

In [ ]:
S = input()
L = [-1]*(ord('z')-ord('a')+1)
for i in range(len(S)):
    idx = ord(S[i])-ord('a')
    if L[idx] == -1:
        L[idx] = i
print(*L)

<ore>
문제
문자열 S를 입력받은 후에, 각 문자를 R번 반복해 새 문자열 P를 만든 후 출력하는 프로그램을 작성하시오. 즉, 첫 번째 문자를 R번 반복하고, 두 번째 문자를 R번 반복하는 식으로 P를 만들면 된다. S에는 QR Code "alphanumeric" 문자만 들어있다.

QR Code "alphanumeric" 문자는 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ\$%*+-./: 이다.

입력
첫째 줄에 테스트 케이스의 개수 T(1 ≤ T ≤ 1,000)가 주어진다. 각 테스트 케이스는 반복 횟수 R(1 ≤ R ≤ 8), 문자열 S가 공백으로 구분되어 주어진다. S의 길이는 적어도 1이며, 20글자를 넘지 않는다. 

출력
각 테스트 케이스에 대해 P를 출력한다.

예제 입력 1 
2
3 ABC
5 /HTP
예제 출력 1 
AAABBBCCC
/////HHHHHTTTTTPPPPP
</ore>

In [ ]:
T = int(input())
for i in range(T):
    R, S = input().split()
    R = int(R)
    str = ''
    for x in range(len(S)):
        str += S[x]*R
    print(str)

<pre>
문제
영어 대소문자와 공백으로 이루어진 문자열이 주어진다. 이 문자열에는 몇 개의 단어가 있을까? 이를 구하는 프로그램을 작성하시오. 단, 한 단어가 여러 번 등장하면 등장한 횟수만큼 모두 세어야 한다.

입력
첫 줄에 영어 대소문자와 공백으로 이루어진 문자열이 주어진다. 이 문자열의 길이는 1,000,000을 넘지 않는다. 단어는 공백 한 개로 구분되며, 공백이 연속해서 나오는 경우는 없다. 또한 문자열은 공백으로 시작하거나 끝날 수 있다.

출력
첫째 줄에 단어의 개수를 출력한다.

예제 입력 1 
The Curious Case of Benjamin Button

예제 출력 1 
6
</pre>

In [ ]:

print(len(list(input().split())))

<pre>
문제
상근이의 동생 상수는 수학을 정말 못한다. 상수는 숫자를 읽는데 문제가 있다. 이렇게 수학을 못하는 상수를 위해서 상근이는 수의 크기를 비교하는 문제를 내주었다. 상근이는 세 자리 수 두 개를 칠판에 써주었다. 그 다음에 크기가 큰 수를 말해보라고 했다.

상수는 수를 다른 사람과 다르게 거꾸로 읽는다. 예를 들어, 734와 893을 칠판에 적었다면, 상수는 이 수를 437과 398로 읽는다. 따라서, 상수는 두 수중 큰 수인 437을 큰 수라고 말할 것이다.

두 수가 주어졌을 때, 상수의 대답을 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 상근이가 칠판에 적은 두 수 A와 B가 주어진다. 두 수는 같지 않은 세 자리 수이며, 0이 포함되어 있지 않다.

출력
첫째 줄에 상수의 대답을 출력한다.

예제 입력 1 
734 893
예제 출력 1 
437
</pre>

In [ ]:
L = list(input().split())
L = [int(''.join(reversed(i))) for i in L]
print(max(L))

In [ ]:
S = 'W'
print(ord(S)-ord('A')+1)

In [ ]:
S = input()
# 할당된 숫자
num = 0
# 소요 시간 (할당된 숫자 + 1)
T = 0
# 한 숫자당 할당된 알파벳 개수 
L = [0, 3, 3, 3, 3, 3, 4, 3, 4]
# 알파벳 개수의 누적합 [0, 3, 6, 9, 12, 15, 19, 22, 26]
sumL = [0]*len(L)
for i in range(1, len(L)):
    sumL[i] = sumL[i-1]+L[i]
    
# 누적합 리스트에서 각 알파벳이 있는 위치를 찾아, 시간을 구함
for i in range(len(S)):
    abc = ord(S[i])-ord('A')+1 # 몇번째 알파벳인지
    for j in range(9):
        if sumL[j] >= abc: # 누적합 리스트에서 몇 번째 인덱스인지 찾기
            num = j+1
            break
    T += num+1
    
print(T)



<pre>
문제
입력 받은 대로 출력하는 프로그램을 작성하시오.

입력
입력이 주어진다. 입력은 최대 100줄로 이루어져 있고, 알파벳 소문자, 대문자, 공백, 숫자로만 이루어져 있다. 각 줄은 100글자를 넘지 않으며, 빈 줄은 주어지지 않는다. 또, 각 줄은 공백으로 시작하지 않고, 공백으로 끝나지 않는다.

출력
입력받은 그대로 출력한다.

예제 입력 1 
Hello
Baekjoon
Online Judge

예제 출력 1 
Hello
Baekjoon
Online Judge
</pre>

In [ ]:
try:
    while True:
        S = input()
        if S == '':
            break
        else:
            print(S)
except:
    None


<pre>
<h3>심화</h3>
         ,r'"7
r`-_   ,'  ,/
 \. ". L_r'
   `~\/
      |
      | 
      출력
</pre>

In [ ]:

print('''         ,r'"7
r`-_   ,'  ,/
 \\. ". L_r'
   `~\\/
      |
      |''')

<pre>
문제
동혁이는 오래된 창고를 뒤지다가 낡은 체스판과 피스를 발견했다.

체스판의 먼지를 털어내고 걸레로 닦으니 그럭저럭 쓸만한 체스판이 되었다. 하지만, 검정색 피스는 모두 있었으나, 흰색 피스는 개수가 올바르지 않았다.

체스는 총 16개의 피스를 사용하며, 킹 1개, 퀸 1개, 룩 2개, 비숍 2개, 나이트 2개, 폰 8개로 구성되어 있다.

동혁이가 발견한 흰색 피스의 개수가 주어졌을 때, 몇 개를 더하거나 빼야 올바른 세트가 되는지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 동혁이가 찾은 흰색 킹, 퀸, 룩, 비숍, 나이트, 폰의 개수가 주어진다. 이 값은 0보다 크거나 같고 10보다 작거나 같은 정수이다.

출력
첫째 줄에 입력에서 주어진 순서대로 몇 개의 피스를 더하거나 빼야 되는지를 출력한다. 만약 수가 양수라면 동혁이는 그 개수 만큼 피스를 더해야 하는 것이고, 음수라면 제거해야 하는 것이다.
</pre>

In [ ]:
# 백준에선 이걸 못쓴다고 한다
import numpy as np
arr1 = np.array([1, 1, 2, 2, 2, 8])
arr2 = np.array(list(map(int, input().split())))
print(*(arr2 - arr1))

In [ ]:
L1 = [1, 1, 2, 2, 2, 8]
L2 = list(map(int, input().split()))
print(*map(lambda i : L1[i]-L2[i], range(len(L2))))

<pre>
문제
예제를 보고 규칙을 유추한 뒤에 별을 찍어 보세요.

입력
첫째 줄에 N(1 ≤ N ≤ 100)이 주어진다.

출력
첫째 줄부터 2×N-1번째 줄까지 차례대로 별을 출력한다.

예제 입력 1 
5
예제 출력 1 
    *
   ***
  *****
 *******
*********
 *******
  *****
   ***
    *
</pre>

In [ ]:
N = int(input())
for i in range(N):
    s = ' '*(N-1-i) + '*'*(i+1) + '*'*i
    print(s)
    
for i in range(N-1):
    s = ' '*(i+1) + '*'*(N-1-i) + '*'*(N-2-i)
    print(s)

In [ ]:
N = 5
n = 9
' '*4 + '*'*1 + '*'*0
' '*3 + '*'*2 + '*'*1
' '*2 + '*'*3 + '*'*2
' '*1 + '*'*4 + '*'*3
' '*0 + '*'*5 + '*'*4

' '*1 + '*'*4 + '*'*3
' '*2 + '*'*3 + '*'*2
' '*3 + '*'*2 + '*'*1
' '*4 + '*'*1 + '*'*0

<pre>
문제
알파벳 소문자로만 이루어진 단어가 주어진다. 이때, 이 단어가 팰린드롬인지 아닌지 확인하는 프로그램을 작성하시오.

팰린드롬이란 앞으로 읽을 때와 거꾸로 읽을 때 똑같은 단어를 말한다. 

level, noon은 팰린드롬이고, baekjoon, online, judge는 팰린드롬이 아니다.

입력
첫째 줄에 단어가 주어진다. 단어의 길이는 1보다 크거나 같고, 100보다 작거나 같으며, 알파벳 소문자로만 이루어져 있다.

출력
첫째 줄에 팰린드롬이면 1, 아니면 0을 출력한다.

예제 입력 1 
level

예제 출력 1
1
</pre>

In [ ]:
str = input()
result = 1
for i in range(len(str)//2):
    if str[i] != str[len(str)-1-i]:
        result = 0
        break
print(result)

<pre>
문제
알파벳 대소문자로 된 단어가 주어지면, 이 단어에서 가장 많이 사용된 알파벳이 무엇인지 알아내는 프로그램을 작성하시오. 단, 대문자와 소문자를 구분하지 않는다.

입력
첫째 줄에 알파벳 대소문자로 이루어진 단어가 주어진다. 주어지는 단어의 길이는 1,000,000을 넘지 않는다.

출력
첫째 줄에 이 단어에서 가장 많이 사용된 알파벳을 대문자로 출력한다. 단, 가장 많이 사용된 알파벳이 여러 개 존재하는 경우에는 ?를 출력한다.

예제 입력 1 
Mississipi
예제 출력 1 
?

예제 입력 2 
zZa
예제 출력 2 
Z
</pre>

In [ ]:
S = input().upper()
L = [0]*(ord('Z')-ord('A')+1) #알파벳이 담길 리스트

for i in range(len(S)) :
    L[ord(S[i])-ord('A')] += 1

M = max(L)
if L.count(M) > 1:
    print('?')
else:
    print(chr(L.index(M)+ord('A')))

<pre>
예전에는 운영체제에서 크로아티아 알파벳을 입력할 수가 없었다. 따라서, 다음과 같이 크로아티아 알파벳을 변경해서 입력했다.

크로아티아 알파벳	변경
č	c=
ć	c-
dž	dz=
đ	d-
lj	lj
nj	nj
š	s=
ž	z=
예를 들어, ljes=njak은 크로아티아 알파벳 6개(lj, e, š, nj, a, k)로 이루어져 있다. 단어가 주어졌을 때, 몇 개의 크로아티아 알파벳으로 이루어져 있는지 출력한다.

dž는 무조건 하나의 알파벳으로 쓰이고, d와 ž가 분리된 것으로 보지 않는다. lj와 nj도 마찬가지이다. 위 목록에 없는 알파벳은 한 글자씩 센다.

입력
첫째 줄에 최대 100글자의 단어가 주어진다. 알파벳 소문자와 '-', '='로만 이루어져 있다.

단어는 크로아티아 알파벳으로 이루어져 있다. 문제 설명의 표에 나와있는 알파벳은 변경된 형태로 입력된다.

출력
입력으로 주어진 단어가 몇 개의 크로아티아 알파벳으로 이루어져 있는지 출력한다.
예제 입력 1 
ljes=njak
예제 출력 1 
6
예제 입력 2 
ddz=z=
예제 출력 2 
3
예제 입력 3 
nljj
예제 출력 3 
3
예제 입력 4 
c=c=
예제 출력 4 
2
예제 입력 5 
dz=ak
예제 출력 5 
3
</pre>

In [ ]:
# 조건식 내에 다음 인덱스를 포함하지 말자. 
# 조건식에서 인덱스 사용 시 언제나 현재 인덱스 사용하기
str = ''.join(reversed(input()))
n = 0
i = 0
for x in range(len(str)):
    if str[i] == '=':
        i += 1
        if str[i] == 'z' and i < len(str)-1:
            if str[i+1] == 'd':
                i += 1
    elif str[i] == '-':
        i += 1
    if str[i] == 'j' and i < len(str)-1:
        if str[i+1] in ['l', 'n']:
            i += 1
    n += 1
    i += 1
    if i > len(str)-1: break
print(n)


<pre>
문제
그룹 단어란 단어에 존재하는 모든 문자에 대해서, 각 문자가 연속해서 나타나는 경우만을 말한다. 예를 들면, ccazzzzbb는 c, a, z, b가 모두 연속해서 나타나고, kin도 k, i, n이 연속해서 나타나기 때문에 그룹 단어이지만, aabbbccb는 b가 떨어져서 나타나기 때문에 그룹 단어가 아니다.

단어 N개를 입력으로 받아 그룹 단어의 개수를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 단어의 개수 N이 들어온다. N은 100보다 작거나 같은 자연수이다. 둘째 줄부터 N개의 줄에 단어가 들어온다. 단어는 알파벳 소문자로만 되어있고 중복되지 않으며, 길이는 최대 100이다.

출력
첫째 줄에 그룹 단어의 개수를 출력한다.

예제 입력 1 
3
happy
new
year
예제 출력 1 
3

예제 입력 2 
4
aba
abab
abcabc
a
예제 출력 2 
1
</pre>

In [ ]:
# 벙법1. 메모리 사용을 줄이되, 사용 시간이 더 걸릴거 같지만, 단어가 등장할 때마다 리스트에 저장 후 매번 있는지 찾기 -> 비효율인ㄴ듯..?
# 방법2. 알파벳 개수만큼의 리스트 구성, 등장 시 해당 값을 표시-> 인덱스로 바로 접근이 가능하여, 비교하는 과정이 없어서 시간 단축 가능
answer = 0

for x in range(int(input())):
    str = input()
    L = [0]*(ord('z') - ord('a') + 1)
    isit = True
    prev = ''
    for i in range(len(str)):
        idx = ord(str[i])-ord('a')
        if L[idx] == 0:
            L[idx] = 1
            prev = str[i]
            continue
        else:
            if str[i] == prev:
                continue
            else:
                isit = False
                break
    if isit:
        answer += 1

print(answer)

<pre>
문제
인하대학교 컴퓨터공학과를 졸업하기 위해서는, 전공평점이 3.3 이상이거나 졸업고사를 통과해야 한다. 그런데 아뿔싸, 치훈이는 깜빡하고 졸업고사를 응시하지 않았다는 사실을 깨달았다!

치훈이의 전공평점을 계산해주는 프로그램을 작성해보자.

전공평점은 전공과목별 (학점 × 과목평점)의 합을 학점의 총합으로 나눈 값이다.

인하대학교 컴퓨터공학과의 등급에 따른 과목평점은 다음 표와 같다.

A+	4.5
A0	4.0
B+	3.5
B0	3.0
C+	2.5
C0	2.0
D+	1.5
D0	1.0
F	0.0
P/F 과목의 경우 등급이 P또는 F로 표시되는데, 등급이 P인 과목은 계산에서 제외해야 한다.

과연 치훈이는 무사히 졸업할 수 있을까?

입력
20줄에 걸쳐 치훈이가 수강한 전공과목의 과목명, 학점, 등급이 공백으로 구분되어 주어진다.

출력
치훈이의 전공평점을 출력한다.

정답과의 절대오차 또는 상대오차가 
\(10^{-4}\) 이하이면 정답으로 인정한다.

제한
1 ≤ 과목명의 길이 ≤ 50
과목명은 알파벳 대소문자 또는 숫자로만 이루어져 있으며, 띄어쓰기 없이 주어진다. 입력으로 주어지는 모든 과목명은 서로 다르다.
학점은 1.0,2.0,3.0,4.0중 하나이다.
등급은 A+,A0,B+,B0,C+,C0,D+,D0,F,P중 하나이다.
적어도 한 과목은 등급이 P가 아님이 보장된다.
예제 입력 1 
ObjectOrientedProgramming1 3.0 A+
IntroductiontoComputerEngineering 3.0 A+
ObjectOrientedProgramming2 3.0 A0
CreativeComputerEngineeringDesign 3.0 A+
AssemblyLanguage 3.0 A+
InternetProgramming 3.0 B0
ApplicationProgramminginJava 3.0 A0
SystemProgramming 3.0 B0
OperatingSystem 3.0 B0
WirelessCommunicationsandNetworking 3.0 C+
LogicCircuits 3.0 B0
DataStructure 4.0 A+
MicroprocessorApplication 3.0 B+
EmbeddedSoftware 3.0 C0
ComputerSecurity 3.0 D+
Database 3.0 C+
Algorithm 3.0 B0
CapstoneDesigninCSE 3.0 B+
CompilerDesign 3.0 D0
ProblemSolving 4.0 P
예제 출력 1 
3.284483
</pre>

In [ ]:
score = ['D', 'C', 'B', 'A']
총학점 = 0
전공평점 = 0
for i in range(20):
    # 전공평점 =  전공과목별 (학점 × 과목평점)의 합을 학점의 총합으로 나눈 값
    입력 = list(input().split())
    ABCDF = 입력[2][0]
    if ABCDF == 'P':
        continue
    elif ABCDF == 'F':
        학점 = int(입력[1][0])
        총학점 += 학점
    else: 
        # 취득 학점의 앞부분 A B C D F P 조회 후, 일치하는 L1의 인덱스를 학점으로 사용 (새부 점수는 y)
        x = score.index(ABCDF) + 1
        y = 0.5 if 입력[2][1] == '+' else 0
        과목평점 = x+y
        학점 = int(입력[1][0])
        총학점 += 학점
        전공평점 += (학점*과목평점)

전공평점 = 전공평점/총학점
print(전공평점)



<pre>
<h2>이차원배열</h2>
문제
N*M크기의 두 행렬 A와 B가 주어졌을 때, 두 행렬을 더하는 프로그램을 작성하시오.

입력
첫째 줄에 행렬의 크기 N 과 M이 주어진다. 둘째 줄부터 N개의 줄에 행렬 A의 원소 M개가 차례대로 주어진다. 이어서 N개의 줄에 행렬 B의 원소 M개가 차례대로 주어진다. N과 M은 100보다 작거나 같고, 행렬의 원소는 절댓값이 100보다 작거나 같은 정수이다.

출력
첫째 줄부터 N개의 줄에 행렬 A와 B를 더한 행렬을 출력한다. 행렬의 각 원소는 공백으로 구분한다.

예제 입력 1 
3 3
1 1 1
2 2 2
0 1 0
3 3 3
4 4 4
5 5 100
예제 출력 1 
4 4 4
6 6 6
5 6 100
</pre>

In [ ]:
N, M = map(int, input().split())
L1 = [0]*N
for i in range(N):
    L1[i] = list(map(int, input().split()))
for i in range(N):
    L2 = list(map(int, input().split()))
    for j in range(M):
        L1[i][j] += L2[j]
    print(*L1[i])

<pre>
입력
첫째 줄부터 아홉 번째 줄까지 한 줄에 아홉 개씩 수가 주어진다. 주어지는 수는 100보다 작은 자연수 또는 0이다.

출력
첫째 줄에 최댓값을 출력하고, 둘째 줄에 최댓값이 위치한 행 번호와 열 번호를 빈칸을 사이에 두고 차례로 출력한다. 최댓값이 두 개 이상인 경우 그 중 한 곳의 위치를 출력한다.

예제 입력 1 
3 23 85 34 17 74 25 52 65
10 7 39 42 88 52 14 72 63
87 42 18 78 53 45 18 84 53
34 28 64 85 12 16 75 36 55
21 77 45 35 28 75 90 76 1
25 87 65 15 28 11 37 28 74
65 27 75 41 7 89 78 64 39
47 47 70 45 23 65 3 41 44
87 13 82 38 31 12 29 29 80
예제 출력 1 
90
5 7
</pre>

In [27]:
totalMax = 0
x, y = 0, 0
for i in range(9):
    row = list(map(int, input().split()))
    rowMax = row
    if totalMax <= rowMax:
        totalMax = rowMax
        x = i + 1
        y = row.index(rowMax) + 1
print(totalMax)
print(x, y)

In [ ]:
totalMax = 0
x, y = 0, 0
for i in range(9):
    row = list(map(int, input().split()))
    rowMax = 0
    rowX, rowY = 0, 0
    for j in range(9):
        if rowMax <= row[j]:
            rowMax = row[j]
            rowX = i + 1 
            rowY = j + 1
    if totalMax <= rowMax:
        totalMax = rowMax
        x = rowX
        y = rowY
print(totalMax)
print(x, y)

<pre>
문제
아직 글을 모르는 영석이가 벽에 걸린 칠판에 자석이 붙어있는 글자들을 붙이는 장난감을 가지고 놀고 있다. 

이 장난감에 있는 글자들은 영어 대문자 ‘A’부터 ‘Z’, 영어 소문자 ‘a’부터 ‘z’, 숫자 ‘0’부터 ‘9’이다. 영석이는 칠판에 글자들을 수평으로 일렬로 붙여서 단어를 만든다. 다시 그 아래쪽에 글자들을 붙여서 또 다른 단어를 만든다. 이런 식으로 다섯 개의 단어를 만든다. 아래 그림 1은 영석이가 칠판에 붙여 만든 단어들의 예이다. 

A A B C D D
a f z z 
0 9 1 2 1
a 8 E W g 6
P 5 h 3 k x
<그림 1>

한 줄의 단어는 글자들을 빈칸 없이 연속으로 나열해서 최대 15개의 글자들로 이루어진다. 또한 만들어진 다섯 개의 단어들의 글자 개수는 서로 다를 수 있다. 

심심해진 영석이는 칠판에 만들어진 다섯 개의 단어를 세로로 읽으려 한다. 세로로 읽을 때, 각 단어의 첫 번째 글자들을 위에서 아래로 세로로 읽는다. 다음에 두 번째 글자들을 세로로 읽는다. 이런 식으로 왼쪽에서 오른쪽으로 한 자리씩 이동 하면서 동일한 자리의 글자들을 세로로 읽어 나간다. 위의 그림 1의 다섯 번째 자리를 보면 두 번째 줄의 다섯 번째 자리의 글자는 없다. 이런 경우처럼 세로로 읽을 때 해당 자리의 글자가 없으면, 읽지 않고 그 다음 글자를 계속 읽는다. 그림 1의 다섯 번째 자리를 세로로 읽으면 D1gk로 읽는다. 

그림 1에서 영석이가 세로로 읽은 순서대로 글자들을 공백 없이 출력하면 다음과 같다:

Aa0aPAf985Bz1EhCz2W3D1gkD6x

칠판에 붙여진 단어들이 주어질 때, 영석이가 세로로 읽은 순서대로 글자들을 출력하는 프로그램을 작성하시오.

입력
총 다섯줄의 입력이 주어진다. 각 줄에는 최소 1개, 최대 15개의 글자들이 빈칸 없이 연속으로 주어진다. 주어지는 글자는 영어 대문자 ‘A’부터 ‘Z’, 영어 소문자 ‘a’부터 ‘z’, 숫자 ‘0’부터 ‘9’ 중 하나이다. 각 줄의 시작과 마지막에 빈칸은 없다.

출력
영석이가 세로로 읽은 순서대로 글자들을 출력한다. 이때, 글자들을 공백 없이 연속해서 출력한다. 

예제 입력 1 
ABCDE
abcde
01234
FGHIJ
fghij
예제 출력 1 
Aa0FfBb1GgCc2HhDd3IiEe4Jj

예제 입력 2 
AABCDD
afzz
09121
a8EWg6
P5h3kx
예제 출력 2 
Aa0aPAf985Bz1EhCz2W3D1gkD6x
</pre>

In [ ]:
N, M = 5, 15
L = ['']*(N*M)
answer = ''
for i in range(N):
    str = input()
    for j in range(M):
        if j < len(str):
            L[i+j*N] = str[j]
        else:
            break
for x in L:
    answer += x
print(answer)


<pre>

문제
가로, 세로의 크기가 각각 100인 정사각형 모양의 흰색 도화지가 있다. 이 도화지 위에 가로, 세로의 크기가 각각 10인 정사각형 모양의 검은색 색종이를 색종이의 변과 도화지의 변이 평행하도록 붙인다. 이러한 방식으로 색종이를 한 장 또는 여러 장 붙인 후 색종이가 붙은 검은 영역의 넓이를 구하는 프로그램을 작성하시오.



예를 들어 흰색 도화지 위에 세 장의 검은색 색종이를 그림과 같은 모양으로 붙였다면 검은색 영역의 넓이는 260이 된다.

입력
첫째 줄에 색종이의 수가 주어진다. 이어 둘째 줄부터 한 줄에 하나씩 색종이를 붙인 위치가 주어진다. 색종이를 붙인 위치는 두 개의 자연수로 주어지는데 첫 번째 자연수는 색종이의 왼쪽 변과 도화지의 왼쪽 변 사이의 거리이고, 두 번째 자연수는 색종이의 아래쪽 변과 도화지의 아래쪽 변 사이의 거리이다. 색종이의 수는 100 이하이며, 색종이가 도화지 밖으로 나가는 경우는 없다

출력
첫째 줄에 색종이가 붙은 검은 영역의 넓이를 출력한다.

예제 입력 1 
3
3 7
15 7
5 2
예제 출력 1 
260
</pre>

In [ ]:
paper = [[0]*100 for i in range(100)]
for n in range(int(input())):
    row, col = map(int, input().split())
    for i in range(10):
        for j in range(10):
            paper[row+i][col+j] = 1 
print(sum(sum(p) for p in paper))


In [ ]:
paper = [[0]*10 for i in range(10)]
row, col = 1,1
for i in range(3):
    for j in range(3):
        paper[row + i][col + j] = 1
    
paper[2][3] = 0

for i in paper:
    print(i)

In [ ]:
N = int(input())
paper = set()
for x in range(N):
    x, y = map(int, input().split())
    for i in range(10):
        for j in range(10):
            paper = paper.union({(x+i, y+j)})
print(len(paper))

In [ ]:
x = [[0]*2]*3
print(x)

In [ ]:
L = [[1, 0, 1], [1, 1, 1], [1, 1, 1]]

print(sum(sum(l) for l in L))

<pre>
문제
B진법 수 N이 주어진다. 이 수를 10진법으로 바꿔 출력하는 프로그램을 작성하시오.

10진법을 넘어가는 진법은 숫자로 표시할 수 없는 자리가 있다. 이런 경우에는 다음과 같이 알파벳 대문자를 사용한다.

A: 10, B: 11, ..., F: 15, ..., Y: 34, Z: 35

입력
첫째 줄에 N과 B가 주어진다. (2 ≤ B ≤ 36)

B진법 수 N을 10진법으로 바꾸면, 항상 10억보다 작거나 같다.

출력
첫째 줄에 B진법 수 N을 10진법으로 출력한다.

예제 입력 1 
ZZZZZ 36
예제 출력 1 
60466175
</pre>

In [9]:
print(ord('9'))
print(ord('A'))
print(int('2'))

57
65
2


In [12]:
# 10진법 : 9까지 이후 자리 수 3자리 수 ABC: 10^2*A+10^1+B+10^0+C
# B진법 : n자리수 : 10^(n-i-1)*str[i]
str, N = input().split()
N = int(N)
answer = 0
for i in range(len(str)):
    x = str[i]
    if ord(x)>=65:
        x = ord(x)-55
    else:
        x = int(x)
    answer += N**(len(str)-i-1)*x
print(answer)


60466175


<pre>
문제
10진법 수 N이 주어진다. 이 수를 B진법으로 바꿔 출력하는 프로그램을 작성하시오.

10진법을 넘어가는 진법은 숫자로 표시할 수 없는 자리가 있다. 이런 경우에는 다음과 같이 알파벳 대문자를 사용한다.

A: 10, B: 11, ..., F: 15, ..., Y: 34, Z: 35

입력
첫째 줄에 N과 B가 주어진다. (2 ≤ B ≤ 36) N은 10억보다 작거나 같은 자연수이다.

출력
첫째 줄에 10진법 수 N을 B진법으로 출력한다.

예제 입력 1 
60466175 36
예제 출력 1 
ZZZZZ
</pre>

In [55]:
import math
N, B = map(int, input().split())
lenB = int(math.log(N)//math.log(B)+1)
answer = ''
for i in range(lenB):
    x = int(N%(B**(lenB-i))//(B**(lenB-i-1)))
    if x > 9:
        x = chr(x+55)
    answer += f'{x}'
print(answer)

ZZZZZ


In [53]:
# 10진법 -> B진법으로 바꾸기
# 몇자리 수인지 어떻게 알아내야하나 혹은 계속 B로 나눈 나머지를 구하고, B^2으로 나눈 나머지를 구하고,,, 반복

import math
# B진법으로 표기했을 때의 자리수
N, B = 60466175, 36
lenB = int(math.log(N)//math.log(B)+1)
answer = ''
for i in range(lenB):
    # 이전 차수의 수로 나눈 나머지를 갖고, 현재 차수의 수가 몇 개 들어있는 지 확인
    x = int(N%(B**(lenB-i))//(B**(lenB-i-1)))
    if x > 9:
        x = chr(x+55)
    answer += f'{x}'
print(answer)

ZZZZZ


<pre>
문제
미국으로 유학간 동혁이는 세탁소를 운영하고 있다. 동혁이는 최근에 아르바이트로 고등학생 리암을 채용했다.

동혁이는 리암에게 실망했다.

리암은 거스름돈을 주는 것을 자꾸 실수한다.

심지어 $0.5달러를 줘야하는 경우에 거스름돈으로 $5달러를 주는것이다!

어쩔수 없이 뛰어난 코딩 실력을 발휘해 리암을 도와주는 프로그램을 작성하려고 하지만, 디아블로를 하느라 코딩할 시간이 없어서 이 문제를 읽고 있는 여러분이 대신 해주어야 한다.

거스름돈의 액수가 주어지면 리암이 줘야할 쿼터(Quarter, $0.25)의 개수, 다임(Dime, $0.10)의 개수, 니켈(Nickel, $0.05)의 개수, 페니(Penny, $0.01)의 개수를 구하는 프로그램을 작성하시오. 거스름돈은 항상 $5.00 이하이고, 손님이 받는 동전의 개수를 최소로 하려고 한다. 예를 들어, $1.24를 거슬러 주어야 한다면, 손님은 4쿼터, 2다임, 0니켈, 4페니를 받게 된다.

입력
첫째 줄에 테스트 케이스의 개수 T가 주어진다. 각 테스트 케이스는 거스름돈 C를 나타내는 정수 하나로 이루어져 있다. C의 단위는 센트이다. (1달러 = 100센트) (1<=C<=500)

출력
각 테스트케이스에 대해 필요한 쿼터의 개수, 다임의 개수, 니켈의 개수, 페니의 개수를 공백으로 구분하여 출력한다.

예제 입력 1 
3
124
25
194
예제 출력 1 
4 2 0 4
1 0 0 0
7 1 1 4
</pre>

In [66]:
"""쿼터(Quarter, $0.25)의 개수, 다임(Dime, $0.10)의 개수, 니켈(Nickel, $0.05)의 개수, 페니(Penny, $0.01)의 개수
거스름돈은 항상 $5.00 이하이고, 손님이 받는 동전의 개수 최소
예를 들어, $1.24를 거슬러 주어야 한다면, 손님은 4쿼터, 2다임, 0니켈, 4페니를 받게 된다.
첫째 줄에 테스트 케이스의 개수 T가 주어진다. 각 테스트 케이스는 거스름돈 C를 나타내는 정수 하나로 이루어져 있다. 
C의 단위는 센트이다. (1달러 = 100센트) (1<=C<=500)"""
# Quarter = 25
# Dime = 10
# Nickel = 5
# Penny = 1
coin = [25, 10, 5, 1]
answer = [0]*4
T = int(input())
for i in range(T):
    C = int(input())
    answer[0] = C//25
    for i in range(1, 4):
        C = C%coin[i-1]
        answer[i] = C//coin[i]
    print(*answer)

4 2 0 4
1 0 0 0
7 1 1 4


<pre>
문제
상근이는 친구들과 함께 SF영화를 찍으려고 한다. 이 영화는 외계 지형이 필요하다. 실제로 우주선을 타고 외계 행성에 가서 촬영을 할 수 없기 때문에, 컴퓨터 그래픽으로 CG처리를 하려고 한다.

외계 지형은 중앙 이동 알고리즘을 이용해서 만들려고 한다.

알고리즘을 시작하면서 상근이는 정사각형을 이루는 점 4개를 고른다. 그 후에는 다음과 같은 과정을 거쳐서 지형을 만든다.

정사각형의 각 변의 중앙에 점을 하나 추가한다.
정사각형의 중심에 점을 하나 추가한다.
초기 상태에서 위와 같은 과정을 한 번 거치면 총 4개의 정사각형이 새로 생긴다. 이와 같은 과정을 상근이가 만족할 때 까지 계속한다.

상근이는 어떤 점은 한 개 보다 많은 정사각형에 포함될 수 있다는 사실을 알았다. 메모리 소모량을 줄이기 위해서 중복하는 점을 한 번만 저장하려고 한다. 과정을 N번 거친 후 점 몇 개를 저장해야 하는지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 N이 주어진다. (1 ≤ N ≤ 15)

출력
첫째 줄에 과정을 N번 거친 후 점의 수를 출력한다.

예제 입력 1 
1
예제 출력 1 
9
예제 입력 2 
2
예제 출력 2 
25
예제 입력 3 
5
예제 출력 3 
1089
</pre>

In [83]:
# 2 3 5 9 과정 포함
# 이전 항 -1을 더해서 제곱 
N = int(input())
sideNode = [3]*(N)
for i in range(1, N):
    sideNode[i] = sideNode[i-1]*2-1 
print(sideNode[-1]**2)

1089


In [203]:
# 답만 얻기 
sideNode = 2
for _ in range(int(input())):
    sideNode = sideNode*2-1
print(sideNode**2)

1089


In [105]:
N = int(input())
sideNode = [3]*N
x=[3]
sideNode += [sideNode[i-1]*2-1 for i in range(1, N)]
print(x)

[3]


<pre>
벌집
위의 그림과 같이 육각형으로 이루어진 벌집이 있다. 그림에서 보는 바와 같이 중앙의 방 1부터 시작해서 이웃하는 방에 돌아가면서 1씩 증가하는 번호를 주소로 매길 수 있다. 숫자 N이 주어졌을 때, 벌집의 중앙 1에서 N번 방까지 최소 개수의 방을 지나서 갈 때 몇 개의 방을 지나가는지(시작과 끝을 포함하여)를 계산하는 프로그램을 작성하시오. 예를 들면, 13까지는 3개, 58까지는 5개를 지난다.

입력
첫째 줄에 N(1 ≤ N ≤ 1,000,000,000)이 주어진다.

출력
입력으로 주어진 방까지 최소 개수의 방을 지나서 갈 때 몇 개의 방을 지나는지 출력한다.

예제 입력 1 
13
예제 출력 1 
3
</pre>

In [219]:
import math
x = math.ceil(1)
print(x)
ceil

1


In [269]:
# 계차수열의 일반항 An = 3n^2+3n+1이 직선 y = k와 만드는 넓이에서 정수 눈금의 좌표를 (n, An)이라고 했을 때, 정수 n의 최대 값
# n = (1/6)*squer_root(12k-3)-1/2의 올림
import math
n = math.sqrt(12*int(input())-3)/6+0.5
print(math.ceil(n))

2


In [286]:
print(int(((12*int(input())-3)**.5)/6+.5))

2


<pre>
문제
무한히 큰 배열에 다음과 같이 분수들이 적혀있다.

1/1	1/2	1/3	1/4	1/5	…
2/1	2/2	2/3	2/4	…	…
3/1	3/2	3/3	…	…	…
4/1	4/2	…	…	…	…
5/1	…	…	…	…	…
…	…	…	…	…	…
이와 같이 나열된 분수들을 1/1 → 1/2 → 2/1 → 3/1 → 2/2 → … 과 같은 지그재그 순서로 차례대로 1번, 2번, 3번, 4번, 5번, … 분수라고 하자.

X가 주어졌을 때, X번째 분수를 구하는 프로그램을 작성하시오.

입력
첫째 줄에 X(1 ≤ X ≤ 10,000,000)가 주어진다.

출력
첫째 줄에 분수를 출력한다.

예제 입력 1 
1
예제 출력 1 
1/1
예제 입력 2 

예제 출력 2 
1/2
예제 입력 3 
3
예제 출력 3 
2/1
예제 입력 4 
4
예제 출력 4 
3/1
예제 입력 5 
5
예제 출력 5 
2/2
예제 입력 6 
6
예제 출력 6 
1/3
예제 입력 7 
7
예제 출력 7 
1/4
예제 입력 8 
8
예제 출력 8 
2/3
예제 입력 9 
9
예제 출력 9 
3/2
예제 입력 10 
14
예제 출력 10 
2/4
</pre>

In [347]:
# 한방향 군 수열
import math
i = 12
k = math.ceil(((1+8*i)**.5-1)/2)
print(k, '번째 그룹')
j = int(i - k*(k-1)/2)
print('그룹 내에서', j, '번째')
print(j, k-j+1)

5 번째 그룹
그룹 내에서 2 번째
2 4


In [358]:
# 지그재그 군 수열
import math
i=int(input())
k=math.ceil(((1+8*i)**.5-1)/2)
j=i-k*(k-1)//2
print(j,'/',k-j+1,sep='') if k%2==0 else print(k-j+1,'/',j,sep='')

2/4


<pre>
문제
땅 위에 달팽이가 있다. 이 달팽이는 높이가 V미터인 나무 막대를 올라갈 것이다.

달팽이는 낮에 A미터 올라갈 수 있다. 하지만, 밤에 잠을 자는 동안 B미터 미끄러진다. 또, 정상에 올라간 후에는 미끄러지지 않는다.

달팽이가 나무 막대를 모두 올라가려면, 며칠이 걸리는지 구하는 프로그램을 작성하시오.

입력
첫째 줄에 세 정수 A, B, V가 공백으로 구분되어서 주어진다. (1 ≤ B < A ≤ V ≤ 1,000,000,000)

출력
첫째 줄에 달팽이가 나무 막대를 모두 올라가는데 며칠이 걸리는지 출력한다.
예제 입력 1 
2 1 5
예제 출력 1 
4
예제 입력 2 
5 1 6
예제 출력 2 
2
예제 입력 3 
100 99 1000000000
예제 출력 3 
999999901
</pre>

In [395]:
# 직선의 방정식으로 해석
import math
A, B, V = map(int, input().split())
print(math.ceil((V-A)/(A-B))+1)

2


<pre>

문제
4 × 3 = 12이다.

이 식을 통해 다음과 같은 사실을 알 수 있다.

3은 12의 약수이고, 12는 3의 배수이다.

4도 12의 약수이고, 12는 4의 배수이다.

두 수가 주어졌을 때, 다음 3가지 중 어떤 관계인지 구하는 프로그램을 작성하시오.

첫 번째 숫자가 두 번째 숫자의 약수이다.
첫 번째 숫자가 두 번째 숫자의 배수이다.
첫 번째 숫자가 두 번째 숫자의 약수와 배수 모두 아니다.
입력
입력은 여러 테스트 케이스로 이루어져 있다. 각 테스트 케이스는 10,000이 넘지않는 두 자연수로 이루어져 있다. 마지막 줄에는 0이 2개 주어진다. 두 수가 같은 경우는 없다.

출력
각 테스트 케이스마다 첫 번째 숫자가 두 번째 숫자의 약수라면 factor를, 배수라면 multiple을, 둘 다 아니라면 neither를 출력한다.

예제 입력 1 
8 16
32 4
17 5
0 0

예제 출력 1 
factor
multiple
neither

</pre>

In [13]:
while True:
    A, B = map(int, input().split())
    if A==0 or B==0: break
    if B%A==0:
        print('factor')
    elif A%B==0:
        print('multiple')
    else:
        print('neither')


<pre>
문제
어떤 자연수 p와 q가 있을 때, 만일 p를 q로 나누었을 때 나머지가 0이면 q는 p의 약수이다. 

6을 예로 들면

6 ÷ 1 = 6 … 0
6 ÷ 2 = 3 … 0
6 ÷ 3 = 2 … 0
6 ÷ 4 = 1 … 2
6 ÷ 5 = 1 … 1
6 ÷ 6 = 1 … 0
그래서 6의 약수는 1, 2, 3, 6, 총 네 개이다.

두 개의 자연수 N과 K가 주어졌을 때, N의 약수들 중 K번째로 작은 수를 출력하는 프로그램을 작성하시오.

입력
첫째 줄에 N과 K가 빈칸을 사이에 두고 주어진다. N은 1 이상 10,000 이하이다. K는 1 이상 N 이하이다.

출력
첫째 줄에 N의 약수들 중 K번째로 작은 수를 출력한다. 만일 N의 약수의 개수가 K개보다 적어서 K번째 약수가 존재하지 않을 경우에는 0을 출력하시오.

예제 입력 1 
6 3
예제 출력 1 
3

예제 입력 2 
25 4
예제 출력 2 
0

예제 입력 3 
2735 1
예제 출력 3 
1
</pre>


In [21]:
N, K = map(int, input().split())
L = []
for i in range(1, N+1):
    if N%i==0:
        L.append(i)
print(0) if len(L) < K else print(L[K-1])
    

1
[1, 5, 547, 2735]


<pre>
문제
어떤 숫자 n이 자신을 제외한 모든 약수들의 합과 같으면, 그 수를 완전수라고 한다.

예를 들어 6은 6 = 1 + 2 + 3 으로 완전수이다.

n이 완전수인지 아닌지 판단해주는 프로그램을 작성하라.

입력
입력은 테스트 케이스마다 한 줄 간격으로 n이 주어진다. (2 < n < 100,000)

입력의 마지막엔 -1이 주어진다.

출력
테스트케이스 마다 한줄에 하나씩 출력해야 한다.

n이 완전수라면, n을 n이 아닌 약수들의 합으로 나타내어 출력한다(예제 출력 참고).

이때, 약수들은 오름차순으로 나열해야 한다.

n이 완전수가 아니라면 n is NOT perfect. 를 출력한다.

예제 입력 1 
6
12
28
-1

예제 출력 1 
6 = 1 + 2 + 3
12 is NOT perfect.
28 = 1 + 2 + 4 + 7 + 14
</pre>

In [33]:
while True:
    N= int(input())
    if N == -1: break
    L = []
    for i in range(1, N+1):
        if N%i==0:
            L.append(i)
    L.pop()
    if sum(x for x in L) == N:
        print(N, '=', ' + '.join(str(i) for i in L))
    else:
        print(N, 'is NOT perfect.')
    


6 = 1 + 2 + 3
12 is NOT perfect.
28 = 1 + 2 + 4 + 7 + 14


<pre>
문제
주어진 수 N개 중에서 소수가 몇 개인지 찾아서 출력하는 프로그램을 작성하시오.

입력
첫 줄에 수의 개수 N이 주어진다. N은 100이하이다. 다음으로 N개의 수가 주어지는데 수는 1,000 이하의 자연수이다.

출력
주어진 수들 중 소수의 개수를 출력한다.

예제 입력 1
4
1 3 5 7

예제 출력 1 
3
</pre>

In [49]:
N = int(input())
L = list(map(int, input().split()))
answer = 0
for l in L:
    n = 0
    for i in range(1, l+1):
        if l%i==0:
            n += 1
        if n > 2:
            break
    if n==2: answer+=1
print(answer)

2


<pre>
문제
자연수 M과 N이 주어질 때 M이상 N이하의 자연수 중 소수인 것을 모두 골라 이들 소수의 합과 최솟값을 찾는 프로그램을 작성하시오.

예를 들어 M=60, N=100인 경우 60이상 100이하의 자연수 중 소수는 61, 67, 71, 73, 79, 83, 89, 97 총 8개가 있으므로, 이들 소수의 합은 620이고, 최솟값은 61이 된다.

입력
입력의 첫째 줄에 M이, 둘째 줄에 N이 주어진다.

M과 N은 10,000이하의 자연수이며, M은 N보다 작거나 같다.

출력
M이상 N이하의 자연수 중 소수인 것을 모두 찾아 첫째 줄에 그 합을, 둘째 줄에 그 중 최솟값을 출력한다. 

단, M이상 N이하의 자연수 중 소수가 없을 경우는 첫째 줄에 -1을 출력한다.

예제 입력 1 
60
100
예제 출력 1 
620
61

예제 입력 2 
64
65
예제 출력 2 
-1
</pre>

In [64]:
M, N = int(input()), int(input())

sum, min = 0, 0
for x in range(N, M-1, -1):
    cnt = 0
    for i in range(1, x+1):
        if x%i==0:
            cnt += 1
        if cnt > 2:
            break
    if cnt==2:
        sum += x
        min = x
print(-1) if sum == 0 else print(sum, min, sep="\n")


-1


<pre>
문제
정수 N이 주어졌을 때, 소인수분해하는 프로그램을 작성하시오.

입력
첫째 줄에 정수 N (1 ≤ N ≤ 10,000,000)이 주어진다.

출력
N의 소인수분해 결과를 한 줄에 하나씩 오름차순으로 출력한다. N이 1인 경우 아무것도 출력하지 않는다.

예제 입력 1 
72
예제 출력 1 
2
2
2
3
3
예제 입력 2 
3
예제 출력 2 
3
예제 입력 3 
6
예제 출력 3 
2
3
예제 입력 4 
2
예제 출력 4 
2
예제 입력 5 
9991
예제 출력 5 
97
103
</pre>

In [90]:
# 에라토스테네스의 체
n = 10000 # 2부터 1000까지의 모든 수에 대하여 소수 판별
array = [True for i in range(n + 1)] # 처음엔 모든 수가 소수(True)인 것으로 초기화(0과 1은 제와)

# 에라토스테네스의 체 알고리즘
for i in range(2, int(n**.5) + 1): # 2부터 n의 제곱근까지의 모든 수를 확인하며
    if array[i] == True: # i가 소수인 경우(남은 수인 경우)
        # i를 제외한 i의 모든 배수를 지우기
        j = 2
        while i * j <= n:
            array[i * j] = False
            j += 1

# 모든 소수 출력
for i in range(2, n + 1):

    if array[i]:
        print(i)


2
3
5
7
11
13
17
19
23
29
31
37
41
43
47
53
59
61
67
71
73
79
83
89
97
101
103
107
109
113
127
131
137
139
149
151
157
163
167
173
179
181
191
193
197
199
211
223
227
229
233
239
241
251
257
263
269
271
277
281
283
293
307
311
313
317
331
337
347
349
353
359
367
373
379
383
389
397
401
409
419
421
431
433
439
443
449
457
461
463
467
479
487
491
499
503
509
521
523
541
547
557
563
569
571
577
587
593
599
601
607
613
617
619
631
641
643
647
653
659
661
673
677
683
691
701
709
719
727
733
739
743
751
757
761
769
773
787
797
809
811
821
823
827
829
839
853
857
859
863
877
881
883
887
907
911
919
929
937
941
947
953
967
971
977
983
991
997
1009
1013
1019
1021
1031
1033
1039
1049
1051
1061
1063
1069
1087
1091
1093
1097
1103
1109
1117
1123
1129
1151
1153
1163
1171
1181
1187
1193
1201
1213
1217
1223
1229
1231
1237
1249
1259
1277
1279
1283
1289
1291
1297
1301
1303
1307
1319
1321
1327
1361
1367
1373
1381
1399
1409
1423
1427
1429
1433
1439
1447
1451
1453
1459
1471
1481
1483
1487
1489
1493
1499
15

In [96]:
n = 8629

L = [1 for i in range(n + 1)] 

for i in range(2, int(n**.5) + 1):
    if L[i]: 
        j = 2
        while i * j <= n:
            L[i * j] = 0
            j += 1

for i in range(2, n+1):
    if L[i]:
        while not n%i:
            print(i)
            n = n/i
    if n == 1: break


8629


In [121]:
N = int(input())
k = 2
for _ in range(N):
    if N%k:
        k += 1
    else:
        print(k)
        N/=k

2
2
17
691
877


<pre>
문제
한수는 지금 (x, y)에 있다. 직사각형은 각 변이 좌표축에 평행하고, 왼쪽 아래 꼭짓점은 (0, 0), 오른쪽 위 꼭짓점은 (w, h)에 있다. 직사각형의 경계선까지 가는 거리의 최솟값을 구하는 프로그램을 작성하시오.

입력
첫째 줄에 x, y, w, h가 주어진다.

출력
첫째 줄에 문제의 정답을 출력한다.

제한
1 ≤ w, h ≤ 1,000
1 ≤ x ≤ w-1
1 ≤ y ≤ h-1
x, y, w, h는 정수
예제 입력 1 
6 2 10 3
예제 출력 1 
1

예제 입력 2 
1 1 5 5
예제 출력 2 
1

예제 입력 3 
653 375 1000 1000
예제 출력 3 
347

예제 입력 4 
161 181 762 375
예제 출력 4 
161
</pre>

In [8]:
x, y, w, h = map(int, input().split())
print(min(x, y, w-x, h-y))

1

<pre>
문제
세 점이 주어졌을 때, 축에 평행한 직사각형을 만들기 위해서 필요한 네 번째 점을 찾는 프로그램을 작성하시오.

입력
세 점의 좌표가 한 줄에 하나씩 주어진다. 좌표는 1보다 크거나 같고, 1000보다 작거나 같은 정수이다.

출력
직사각형의 네 번째 점의 좌표를 출력한다.

예제 입력 1 
5 5
5 7
7 5
예제 출력 1 
7 7
예제 입력 2 
30 20
10 10
10 20
예제 출력 2 
30 10
</pre>

In [ ]:
X, Y = [], []
for i in range(3):
    x1, y1 = map(int, input().split())
    X.append(x1)
    Y.append(y1)

print(*[x for x in X if X.count(x)==1], end=' ')
print(*[y for y in Y if Y.count(y)==1])

In [29]:
x=y=0
exec('a,b=map(int,input().split());x^=a;y^=b;'*3)
print(x,y)


3 3


<pre>
입력
첫째 줄에는 점의 개수 N (1 ≤ N ≤ 100,000) 이 주어진다. 이어지는 N 줄에는 각 점의 좌표가 두 개의 정수로 한 줄에 하나씩 주어진다. 각각의 좌표는 -10,000 이상 10,000 이하의 정수이다. 

출력
첫째 줄에 N 개의 점을 둘러싸는 최소 크기의 직사각형의 넓이를 출력하시오. 

예제 입력 1 
3
20 24
40 21
10 12
예제 출력 1 
360

예제 입력 2 
1
15 13
예제 출력 2 
0

예제 입력 3 
4
2 1
3 2
5 2
3 4
예제 출력 3 
9
</pre>

In [4]:
X, Y = [], []
for i in range(int(input())):
    x1, y1 = map(int, input().split())
    X.append(x1)
    Y.append(y1)
print((max(X)-min(X))*(max(Y)-min(Y)))

9


<pre>
세 각의 크기가 모두 60이면, Equilateral
세 각의 합이 180이고, 두 각이 같은 경우에는 Isosceles
세 각의 합이 180이고, 같은 각이 없는 경우에는 Scalene
세 각의 합이 180이 아닌 경우에는 Error

입력
총 3개의 줄에 걸쳐 삼각형의 각의 크기가 주어진다. 모든 정수는 0보다 크고, 180보다 작다.

출력
문제의 설명에 따라 Equilateral, Isosceles, Scalene, Error 중 하나를 출력한다.

예제 입력 1 
60
70
50

예제 출력 1 
Scalene
</pre>

In [67]:
A = [int(input()) for _ in range(3)]
if sum(A) == 180:
    cnt = A.count(A[0]^A[1]^A[2])
    if cnt == 3: 
        print('Equilateral')
    elif cnt == 1:
        print('Isosceles')
    else:
        print('Scalene')
else:
    print('Error')

Error


In [64]:
"""
open(0):
    이 부분은 표준 입력에서 데이터를 읽어오는 역할을 합니다.
    open(0)은 표준 입력을 열어서 입력을 읽어오는데, 예를 들어 파일에서 입력을 받을 때도 사용할 수 있습니다.
    이 코드에서는 open(0)이 한 줄에 세 변의 길이를 공백으로 구분하여 입력받아야 합니다.
    
map(int, open(0)):
    map은 함수(int)와 반복 가능한 객체(open(0)의 출력)을 받아서, 각 요소에 함수 int를 적용하여 반환합니다.
    여기서는 open(0)으로 입력받은 값을 각 숫자로 변환하여 리스트로 만듭니다. 예를 들어 "60 60 60"을 입력받으면 [60, 60, 60]이라는 리스트가 됩니다.
    
[*map(int, open(0))]:
    대괄호 [*...]를 사용하여 map 객체를 리스트로 만듭니다. 
    [*map(int, open(0))]는 [60, 60, 60]처럼 변환된 세 변의 길이를 리스트로 만들어 c 변수에 저장합니다.


왈러스 연산자 := 
    왈러스 연산자 :=는 표현식에서 변수를 선언하고 동시에 할당하는 연산자입니다.
    여기서는 c := [...]로 c에 리스트를 할당하고 sum 함수 내에서 이를 바로 사용합니다.

{*c}는 c의 요소들을 집합(set)으로 변환합니다.
"""

print(['Error','Equilateral','Isosceles','Scalene'][sum(c:=[*map(int,open(0))]==180)*len({*c})])

0


<pre>
문제
삼각형의 세 변의 길이가 주어질 때 변의 길이에 따라 다음과 같이 정의한다.

Equilateral :  세 변의 길이가 모두 같은 경우
Isosceles : 두 변의 길이만 같은 경우
Scalene : 세 변의 길이가 모두 다른 경우
단 주어진 세 변의 길이가 삼각형의 조건을 만족하지 못하는 경우에는 "Invalid" 를 출력한다. 예를 들어 6, 3, 2가 이 경우에 해당한다. 가장 긴 변의 길이보다 나머지 두 변의 길이의 합이 길지 않으면 삼각형의 조건을 만족하지 못한다.

세 변의 길이가 주어질 때 위 정의에 따른 결과를 출력하시오.

입력
각 줄에는 1,000을 넘지 않는 양의 정수 3개가 입력된다. 마지막 줄은 0 0 0이며 이 줄은 계산하지 않는다.

출력
각 입력에 맞는 결과 (Equilateral, Isosceles, Scalene, Invalid) 를 출력하시오.

예제 입력 1 
7 7 7
6 5 4
3 2 5
6 2 6
0 0 0
예제 출력 1 
Equilateral
Scalene
Invalid
Isosceles
</pre>

In [91]:
A = [*map(int, input().split())]

In [88]:
print(A)
print(max(A))
print(sum(A)>2*max(A))
print(sum(A)>2*max(A))



[1, 3, 7]
7
False
0
0


In [97]:
1|2|3

3

In [96]:
while 1:
    S = sum(A:=[*map(int, input().split())])
    if S == 0: break
    if S > 2*max(A):
        cnt = A.count(A[0]^A[1]^A[2])
        if cnt == 3: 
            print('Equilateral')
        elif cnt == 1:
            print('Isosceles')
        else:
            print('Scalene')
    else:
        print('Invalid')

In [9]:
while sum(A:=[*map(int,input().split())])!=0:print(['Invalid','Equilateral','Isosceles','Scalene'][(sum(A)>2*max(A))*len({*A})])

Invalid


<pre>
문제
영선이는 길이가 a, b, c인 세 막대를 가지고 있고, 각 막대의 길이를 마음대로 줄일 수 있다.

영선이는 세 막대를 이용해서 아래 조건을 만족하는 삼각형을 만들려고 한다.

각 막대의 길이는 양의 정수이다
세 막대를 이용해서 넓이가 양수인 삼각형을 만들 수 있어야 한다.
삼각형의 둘레를 최대로 해야 한다.
a, b, c가 주어졌을 때, 만들 수 있는 가장 큰 둘레를 구하는 프로그램을 작성하시오. 

입력
첫째 줄에 a, b, c (1 ≤ a, b, c ≤ 100)가 주어진다.

출력
첫째 줄에 만들 수 있는 가장 큰 삼각형의 둘레를 출력한다.

예제 입력 1 
1 2 3
예제 출력 1 
5
예제 입력 2 
2 2 2
예제 출력 2 
6
예제 입력 3 
1 100 1
예제 출력 3 
3
예제 입력 4 
41 64 16
예제 출력 4 
113
</pre>

In [40]:
A = [*sorted(map(int, input().split()))]
while (S:=sum(A)) <= A[-1]*2:
    A[-1] -= 1
print(S)

3


In [48]:
n = int(input())
print(n*(n-1)*(n-2)//6) if n >2 else print(0)
print(3)

35
3


<pre>
입력
첫째 줄에 함수 f(n)을 나타내는 정수 a1, a0가 주어진다. (0 ≤ |ai| ≤ 100)

다음 줄에 양의 정수 c가 주어진다. (1 ≤ c ≤ 100)

다음 줄에 양의 정수 n0가 주어진다. (1 ≤ n0 ≤ 100)

출력
f(n), c, n0가 O(n) 정의를 만족하면 1, 아니면 0을 출력한다.

예제 입력 1 
7 7
8
1
예제 출력 1 
0
f(n) = 7n + 7, g(n) = n, c = 8, n0 = 1이다. f(1) = 14, c × g(1) = 8이므로 O(n) 정의를 만족하지 못한다.

예제 입력 2 
7 7
8
10
예제 출력 2 
1
f(n) = 7n + 7, g(n) = n, c = 8, n0 = 10이다. 모든 n ≥ 10에 대하여 7n + 7 ≤ 8n 이므로 O(n) 정의를 만족한다.</pre>


In [66]:
a1, a0 = map(int, input().split())
c = int(input())
n0 = int(input())

if a1 == c:
    if a0 <= 0:
        print(1)
    else:
        print(0)
elif a1 < c:
    if a0/(c-a1) <= n0:
        print(1)
    else:
        print(0)
else:
    print(0)

0
